staging/src/k8s.io/code-generator/cmd/api-linter/linters/whitelist.csv

```diff
@@ -0,0 +1,1019 @@
+optional_fields_linter_pointers,io.k8s.apimachinery.pkg.runtime.TypeMeta,APIVersion
```

Comment:
I think this is highlighting how "should be a pointer" is silly and not delivering value.  As I look through this list, I wouldn't make most of these pointers.  Why have a linter that checks things that don't make sense?

Reply:
+ @bgrant0607

Are you suggesting we should change our [API convention](https://github.com/kubernetes/community/blob/master/contributors/devel/api-conventions.md#optional-vs-required) for optional fields?  I think that should be discussed in another thread (probably in sig-architecture). Meanwhile do you have any suggestion for another simpler linter that we can add here to make sure the linter framework works?

Reply:
First, our written conventions should reflect our actual conventions.

Second, I don't care how it's implemented, but we need to be able to uniformly distinguish set from unset fields, regardless of language-default values, and to omit unspecified fields when encoding resources that have not had defaults expanded. Pointers were a means to that end. If we can achieve it in a simpler manner, I'd be happy with that.